src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -777,8 +777,9 @@
       if (excludeBaseClasses && class %in% c("list", "environment"))
          next
       
-      methodName <- paste(".DollarNames", class, sep = ".")
-      method <- .rs.getAnywhere(methodName, envir = envir)
+      method <- utils::getS3method(".DollarNames", class, optional = TRUE, 
+        envir = getNamespace("utils"))
```

Comment:
Actually, wouldn't we want to change the whole method definition here? Right now we're iterating over the classes of the object; however, we likely just want to directly call `utils::getS3method` and ignore the class iteration bit.

I think we'd still want to screen out `list` and `environment` classes just to preserve pre-existing behavior.